sorts/bucket_sort.py

```diff
@@ -10,38 +10,25 @@
 # and therefore can also be considered a comparison sort algorithm. The computational complexity estimates
 # involve the number of buckets.
 
-#  Time Complexity of Solution:
-#  Best Case O(n); Average Case O(n); Worst Case O(n)
+# Time Complexity of Solution:
+# Best Case O(n); Average Case O(n); Worst Case O(n)
 
 DEFAULT_BUCKET_SIZE=5
-def bucket_sort(my_list,bucket_size=DEFAULT_BUCKET_SIZE):
-    if(my_list==0):
-        print("you don't have any elements in array!")
+def bucket_sort(my_list, bucket_size=DEFAULT_BUCKET_SIZE):
+    if len(my_list) == 0:
+        raise Exception("you don't have any elements in array!")
 
+    min_value, max_value = (min(my_list), max(my_list))
+    bucket_count = ((max_value-min_value) // bucket_size+1)
+    buckets = [[] for _ in range(bucket_count)]
 
-    min_value=min(my_list)
-    max_value=max(my_list)
-
-    bucket_count=(max_value-min_value)//bucket_size+1
-    buckets=[]
-    for i in range(bucket_count):
-        buckets.append([])
     for i in range(len(my_list)):
-        buckets[(my_list[i]-min_value)//bucket_size].append(my_list[i])
-
-
-    sorted_array=[]
-    for i in range(len(buckets)):
-        buckets[i].sort()
-        for j in range(len(buckets[i])):
-            sorted_array.append(buckets[i][j])
-    return sorted_array
-
-
+        buckets[(my_list[i] - min_value) // bucket_size].append(my_list[i])
 
+    return sorted([buckets[i][j] for i in range(len(buckets))
+                                 for j in range(len(buckets[i]))])
 
-#test
-#best on python 3.7.3
-user_input =input('Enter numbers separated by a comma:').strip()
-unsorted =[int(item) for item in user_input.split(',')]
-print(bucket_sort(unsorted))
+if __name__ == "__main__":
+    user_input = input('Enter numbers separated by a comma:').strip()
+    unsorted = [int(n) for n in user_input.split(',') if len(user_input) > 0]
```

Comment:
**Should we add support for float numbers ?** In my opinion, I think that we should ! :+1:  :100: